feat: LimitedInput component

ndv99 opened the pull request:
## Done
- Created "LimitedInput" component

<!--
- Itemised list of what was changed by this PR.
-->

## QA steps

- [ ] Ensure the component is displayed correctly across all breakpoints
- [ ] Ensure the `immutableText` cannot be modified by typing in the input field
- [ ] Ensure the input still works as intended
- [ ] Ensure the element positioning is still correct when a label is not present

## Screenshots
![image](https://github.com/canonical/maas-react-components/assets/35104482/3f9aac1c-300d-47e9-8a05-5a27a6421b28)

<!--
Attach any screenshots or videos that help illustrate or demonstrate the changes made in this PR.
-->

## Notes

To see this working properly, you'll need to click on "Example", and not "Docs". For some reason, the useEffect hook doesn't seem to work on this page.

<!--
(Optional)
Leave any additional notes for the reviewer here.
-->

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://maas-react-components-134.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/start-demo/19575)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

ndv99 commented on src/lib/elements/LimitedInput/LimitedInput.scss:
```diff
@@ -0,0 +1,15 @@
+    padding-left: $spv--small;
+    padding-bottom: calc(0.4rem - 1px);
+    padding-top: calc(0.4rem - 1px);
+    content: var(--immutable, "");
+    top: var(--top, "inherit");
```
I'm aware that this isn't a perfect solution, but for now it's workable as advised by @petermakowski. I'll create a jira ticket to investigate this implementation and find something better, but for now this should unblock https://github.com/canonical/maas-ui/pull/5420

ndv99 commented:
I've discovered a bug when using this in `maas-ui` where the `immutableText` only shows up in the form *after* one of the fields has been focused and blurred. Investigating now

ndv99 commented:
> I've discovered a bug when using this in `maas-ui` where the `immutableText` only shows up in the form _after_ one of the fields has been focused and blurred. Investigating now

I've fixed this now, which also fixed the immutable text not showing in the component docs.

ndv99 commented:
There's some visual issues that only occur when viewing this component in storybook that can't be replicated if this component is implemented directly in maas-ui. You can replicate this by maximising your browser window, and the placeholder text should seem misaligned. Opening and closing dev tools also causes misalignment, and importing this component into maas-ui also causes the `immutableText` to not register on first render.

For now, I'm implementing this component directly in maas-ui, but I've created a jira ticket to eventually upstream it to maas-react-components and fix these bugs. For now though, this has been blocking the reserved IPs work, so I'm putting this PR on hold
